keel-core/src/main/kotlin/com/netflix/spinnaker/keel/core/api/EnvironmentArtifactPin.kt

```diff
@@ -15,7 +14,10 @@ data class PinnedEnvironment(
   val deliveryConfigName: String,
   val targetEnvironment: String,
   val artifact: DeliveryArtifact,
-  val version: String
+  val version: String,
+  val pinnedBy: String?,
+  val pinnedAt: Long?,
+  val comment: String?
 )
```

Comment:
Adding more data here so that we could maybe display this info (not now, I definitely didn't expose this in the UI at all) (but @erikmunson maybe you could tell me if you even want this data about a pinned artifact and if so where you want it)

Reply:
This is all excellent and I would love to see every field make it to the client! Lines it up very nicely to show a card like we do for judgements:

<img width="877" alt="Screen Shot 2020-04-13 at 7 06 55 PM" src="https://user-images.githubusercontent.com/1850998/79178390-fc16dd00-7db9-11ea-9cdb-9c9587674544.png">

Reply:
I love this field addition! I think it's valuable and I even for doing it mandatory... since if the user is pinning some artifact, shouldn't he explain why for others to see?

Reply:
Ok I'll send it to you.

Reply:
@gal-yardeni re: making a comment mandatory - while I think that is nice in theory, I think it would be full of friction in practice. I think that's more of a culture thing w/in teams to enforce. For example, if it's just you maintaining an app, it would be so annoying to have to fill in the comment every time.

Reply:
@emjburns Yeah I think you're right :) We should though highly encourage people to explain why there are pinning it..